### PR TITLE
1.On plugin error, exit session 2.Message for user in the beginning of the program

### PR DESCRIPTION
**Pull request by shaiRubin770**

### CODE REVIEW

File: Infrastructure/PluginExecutor.cs [hunk 1]
```diff
@@ -112,9 +113,13 @@ private void Execute(string pluginId, string input, string user)
                 plugin.Execute(new PluginInput(input, persistentData, callbacks));
 
             }
-            catch (Exception)
+            catch (Exception ex)
             {
-                _messageSender.SendMessage(user, "An error occured while executing the plugin.");
+                _messageSender.SendMessage(user, "An error occured while executing the plugin, please type help again");
+                var plugin= new PluginInput(input, null, callbacks);
+                plugin.Callbacks.EndSession();
```
Comment (YacovGold) [inline Infrastructure/PluginExecutor.cs:123]: Please format

File: Infrastructure/PluginExecutor.cs [hunk 2]
```diff
@@ -112,9 +113,13 @@ private void Execute(string pluginId, string input, string user)
                 plugin.Execute(new PluginInput(input, persistentData, callbacks));
 
             }
-            catch (Exception)
+            catch (Exception ex)
```
Comment (YacovGold) [inline Infrastructure/PluginExecutor.cs]: No use to ex, please remove

File: Infrastructure/PluginExecutor.cs [hunk 3]
```diff
@@ -112,9 +113,13 @@ private void Execute(string pluginId, string input, string user)
...
             {
-                _messageSender.SendMessage(user, "An error occured while executing the plugin.");
+                _messageSender.SendMessage(user, "An error occured while executing the plugin, please type help again");
+                var plugin= new PluginInput(input, null, callbacks);
+                plugin.Callbacks.EndSession();
+
+
```
Comment (YacovGold) [inline Infrastructure/PluginExecutor.cs]: Extra enters

File: Infrastructure/PluginsManager.cs [hunk 4]
```diff
@@ -28,6 +28,65 @@ public class PluginsManager
...
+            else if (id == Date._Id)
+            {
+                return new Date();
+            }
+            else if (id == NumerologyCalculatorPlugin._Id)
+            {
+                return new NumerologyCalculatorPlugin();
+            }
+
+            else if (id == IsPrimeNumberPlugin._Id)
```
Comment (YacovGold) [inline Infrastructure/PluginsManager.cs]: This is irrelevant  
Please use the new way

File: Plugins/IsPrimeNumber/IsPrimeNumberPlugin.cs [hunk 5]
```diff
@@ -25,19 +25,23 @@ public void Execute(PluginInput input)
             }
             else if(int.TryParse(input.Message,out int num1))
             {
+
```
Comment (YacovGold) [inline Plugins/IsPrimeNumber/IsPrimeNumberPlugin.cs]: Extra enters

File: Plugins/IsPrimeNumber/IsPrimeNumberPlugin.cs [hunk 6]
```diff
@@ -25,19 +25,23 @@ public void Execute(PluginInput input)
...
+                        input.Callbacks.SendMessage(num1 + " is a prime number");
 
                 }
 
             }
             else
             {
-                input.Callbacks.SendMessage("try again, enter a number");
+                int.Parse(input.Message);
+                //input.Callbacks.SendMessage("try again, enter a number");
```
Comment (YacovGold) [inline Plugins/IsPrimeNumber/IsPrimeNumberPlugin.cs]: No need to comments

File: Runners/ConsoleApp/Program.cs [hunk 7]
```diff
@@ -22,8 +22,13 @@ static void Main()
 
         void Main(IMessageSender messageSender)
         {
+
```
Comment (YacovGold) [inline Runners/ConsoleApp/Program.cs]: Extra enters

File: Resource/Resource1.resx [hunk 8]
```diff
@@ -0,0 +1,135 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<root>
```
Comment (YacovGold) [inline Resource/Resource1.resx]: Please remove the resource files

File: Infrastructure/PluginExecutor.cs [hunk 9]
```diff
@@ -112,9 +113,13 @@ private void Execute(string pluginId, string input, string user)
                 plugin.Execute(new PluginInput(input, persistentData, callbacks));
 
             }
-            catch (Exception)
+            catch (Exception ex)
             {
-                _messageSender.SendMessage(user, "An error occured while executing the plugin.");
+                _messageSender.SendMessage(user, "An error occured while executing the plugin, please type help again");
+                var plugin= new PluginInput(input, null, callbacks);
```
Comment (YacovGold) [inline Infrastructure/PluginExecutor.cs]: Please format code

File: Infrastructure/PluginExecutor.cs (hunk 3 above)
Comment (YacovGold) [inline Infrastructure/PluginExecutor.cs]: extra enters

File: Runners/ConsoleApp/Program.cs [hunk 10]
```diff
@@ -22,8 +22,13 @@ static void Main()
...
+            var pluginExecutor = new PluginExecutor(messageSender, new DbDal(), new PluginsMenu(), new PluginsManager());
+            Console.WriteLine("You can type help to get all the pulgins: ");
+
             var pluginExecutor = new PluginExecutor(messageSender, new MemoryDal(), new PluginsMenu(), new PluginsManager());
 
+
```
Comment (YacovGold) [inline Runners/ConsoleApp/Program.cs]: Extra enter

File: Infrastructure/PluginExecutor.cs (hunk 2 above)
Comment (shaiRubin770) [reply Infrastructure/PluginExecutor.cs]: ok,fixed

File: Infrastructure/PluginExecutor.cs (hunk 3 above)
Comment (shaiRubin770) [reply Infrastructure/PluginExecutor.cs]: ok,fixed

File: Plugins/IsPrimeNumber/IsPrimeNumberPlugin.cs (hunk 5 above)
Comment (shaiRubin770) [reply Plugins/IsPrimeNumber/IsPrimeNumberPlugin.cs]: ok,fixed

File: Plugins/IsPrimeNumber/IsPrimeNumberPlugin.cs (hunk 6 above)
Comment (shaiRubin770) [reply Plugins/IsPrimeNumber/IsPrimeNumberPlugin.cs]: ok,fixed

File: Runners/ConsoleApp/Program.cs (hunk 7 above)
Comment (shaiRubin770) [reply Runners/ConsoleApp/Program.cs]: ok,fixed

File: Resource/Resource1.resx (hunk 8 above)
Comment (shaiRubin770) [reply Resource/Resource1.resx]: ok,fixed

File: Infrastructure/PluginExecutor.cs (hunk 3 above)
Comment (shaiRubin770) [reply Infrastructure/PluginExecutor.cs]: ok,fixed

File: Runners/ConsoleApp/Program.cs (hunk 10 above)
Comment (shaiRubin770) [reply Runners/ConsoleApp/Program.cs]: ok,fixed

File: Infrastructure/PluginExecutor.cs (hunk 1 above)
Comment (YacovGold) [reply Infrastructure/PluginExecutor.cs:123]: You can format the code, look here how
https://docs.microsoft.com/en-us/answers/questions/550115/automatic-format-code-in-visual-studio.html

**Pull request merged**